### PR TITLE
SSCS-8215 Updated address regex

### PR DESCRIPTION
**Pull request by danhamilton**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: steps/representative/representative-details/RepresentativeDetails.js [hunk 1]
```diff
@@ -105,22 +106,31 @@ class RepresentativeDetails extends SaveToDraftStore {
...
-          Joi.string().regex(whitelist).allow('')
+          Joi.string().regex(whitelistNotFirst).allow('')
         ) },
       { name: this.pcl.fieldMap.town,
         validator: text.joi(
           fields.townCity.error.required,
-          Joi.string().regex(whitelist).required()
+          Joi.string().regex(whitelistNotFirst).required()
+        ).joi(
+          fields.townCity.error.required,
```
Comment (srochani) [inline steps/representative/representative-details/RepresentativeDetails.js:124]: Will it require invalid here for townCity, as its looks same

File: test/unit/steps/appointee/appointee-contact-details/Appointee-contact-details.test.js [hunk 2]
```diff
@@ -220,8 +220,8 @@ describe('Appointee-contact-details.js', () => {
           expect(field.constructor.name).to.eq('FieldDescriptor');
         });
 
-        it('contains validation', () => {
-          expect(field.validations).to.not.be.empty;
+        it('addressLine2 contains validation', () => {
+          expect(field.validations.length).to.eq(1);
```
Comment (srochani) [inline test/unit/steps/appointee/appointee-contact-details/Appointee-contact-details.test.js:224]: why here length is 1? its not consistent length check 2

File: test/unit/steps/identity/appellant-contact-details/AppellantContactDetails.test.js [hunk 3]
```diff
@@ -216,7 +216,7 @@ describe('AppellantContactDetails.js', () => {
         });
 
         it('contains validation', () => {
-          expect(field.validations).to.not.be.empty;
+          expect(field.validations.length).to.eq(1);
```
Comment (srochani) [inline test/unit/steps/identity/appellant-contact-details/AppellantContactDetails.test.js:219]: same here

File: test/unit/steps/representative/representative-details/RepresentativeDetails.test.js [hunk 4]
```diff
@@ -237,7 +237,7 @@ describe('RepresentativeDetails.js', () => {
         });
 
         it('contains validation', () => {
-          expect(field.validations).to.not.be.empty;
+          expect(field.validations.length).to.eq(1);
```
Comment (srochani) [inline test/unit/steps/representative/representative-details/RepresentativeDetails.test.js:240]: same here

File: test/unit/steps/appointee/appointee-contact-details/Appointee-contact-details.test.js (hunk 2 above)
Comment (danhamilton) [reply test/unit/steps/appointee/appointee-contact-details/Appointee-contact-details.test.js:224]: No, address line 2 is optional and therefore only has 1 validator

File: test/unit/steps/identity/appellant-contact-details/AppellantContactDetails.test.js (hunk 3 above)
Comment (danhamilton) [reply test/unit/steps/identity/appellant-contact-details/AppellantContactDetails.test.js:219]: as above

File: test/unit/steps/representative/representative-details/RepresentativeDetails.test.js (hunk 4 above)
Comment (danhamilton) [reply test/unit/steps/representative/representative-details/RepresentativeDetails.test.js:240]: as above

File: steps/representative/representative-details/RepresentativeDetails.js (hunk 1 above)
Comment (danhamilton) [reply steps/representative/representative-details/RepresentativeDetails.js]: fixed

**Review by srochani — APPROVED** [review]

**Pull request merged**